kms/uri/uri.go

```diff
@@ -82,5 +86,51 @@ func ParseWithScheme(scheme, rawuri string) (*URI, error) {
 // Get returns the first value in the uri with the give n key, it will return
 // empty string if that field is not present.
 func (u *URI) Get(key string) string {
-	return u.Values.Get(key)
+	v := u.Values.Get(key)
+	if v == "" {
+		v = u.URL.Query().Get(key)
+	}
+	return v
+}
+
+// GetEncoded returns the first value in the uri with the give n key, it will
+// return empty nil if that field is not present or is empty. If the return
+// value is hex encoded it will decode it and return it.
+func (u *URI) GetEncoded(key string) []byte {
+	v := u.Get(key)
+	if v == "" {
+		return nil
+	}
+	if len(v)%2 == 0 {
```

Comment:
How does `len(v)%2` verify if the value is hexadecimal?

Reply:
Well, this enforces the value to be like `0A` instead of `A`, `hex.DecodeString()` will return an error if you just pass `A`, so at the end of the day we'll get the same result.

Reply:
What if the value was `foop` (or something with an even number of chars but not intended to be hexadecimal)? Wouldn't be error when trying to decode it? Should we be trying to decode it under those circumstances?